freeipa/src/test/java/com/sequenceiq/freeipa/service/freeipa/backup/cloud/CloudBackupFolderResolverServiceTest.java

```diff
@@ -0,0 +1,103 @@
+package com.sequenceiq.freeipa.service.freeipa.backup.cloud;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.sequenceiq.cloudbreak.auth.altus.CrnParseException;
+import com.sequenceiq.cloudbreak.telemetry.fluent.FluentClusterType;
+import com.sequenceiq.common.api.cloudstorage.old.AdlsGen2CloudStorageV1Parameters;
+import com.sequenceiq.common.api.cloudstorage.old.S3CloudStorageV1Parameters;
+import com.sequenceiq.freeipa.api.model.Backup;
+
+public class CloudBackupFolderResolverServiceTest {
```

Comment:
I think `S3BackupConfigGenerator` and `AdlsGen2BackupConfigGenerator` should be tested in different test, and here you should test the logic related to `CloudBackupFolderResolverService`. I have a feeling that not all the cases are covered here, and it would be too complex to test it through this class

Reply:
Fair enough.   don.